pkg/updater/github.go

```diff
@@ -303,37 +301,39 @@ func (g *Github) DownloadRelease(ctx context.Context, r *github.RepositoryReleas
 	}
 	trace.EndCall(traceCtx)
 
-	file, err = g.processArchive(ctx, file, tmpDir, execName, asset)
-	return file, cleanupFn, err
+	assetPath, err = g.processArchive(ctx, assetPath, tmpDir, execName, asset)
+	return assetPath, cleanupFn, err
 }
 
 // processArchive checks if the file is an archive, and extracts it looking for the same file
 // as the current executable. If it is not an archive then the file given is returned
 func (g *Github) processArchive(ctx context.Context, file, tmpDir, execName string, asset *github.ReleaseAsset) (string, error) {
-	// TODO: better support for other archives in the future
-	if strings.HasSuffix(asset.GetName(), ".tar.gz") {
-		f, err := os.Open(file)
-		if err != nil {
-			return "", errors.Wrap(err, "failed to open downloaded archive")
-		}
-		defer f.Close()
+	if !strings.Contains(asset.GetName(), ".tar") {
```

Comment:
Nit: doing `!strings.HasSuffix(asset.GetName(), ".tar.gz") && !strings.HasSuffix(asset.GetName(), ".tar.xz")` seems safer.

Reply:
I wanted to have less statements, and then error in the handler later on so this was intentional. The actual decompression function handles unknown formats